siriuspy/siriuspy/devices/orbit_interlock.py

```diff
@@ -1,24 +1,149 @@
-"""."""
+"""Orbit Interlock devices."""
 
 import numpy as _np
 
-from .device import Device as _Device
 from ..search import BPMSearch as _BPMSearch
+from ..util import ClassProperty as _classproperty
+from .device import Device as _Device, Devices as _Devices
 
 
-class BPMInterlock(_Device):
-    """BPM Device"""
+class BaseOrbitIntlk:
+    """Base BPM Orbit Interlock Info."""
+
+    CONV_NM2UM = 1e-3  # [nm] --> [um]
+
+    UP_2_DOWN = {
+        'M1': 'M2',
+        'C1-1': 'C1-2',
+        'C2': 'C3-1',
+        'C3-2': 'C4',
+    }
+    DOWN_2_UP = {val: key for key, val in UP_2_DOWN.items()}
+
+    __BPM_NAMES = None
+    __BPM_NICKNAMES = None
+    __BPM_POS = None
+    __BPM_DOWNSTREAM = None
+    __BPM_UPSTREAM = None
+
+    @_classproperty
+    def BPM_NAMES(cls):
+        """BPM names."""
+        if cls.__BPM_NAMES is not None:
+            return cls.__BPM_NAMES
+
+        cls.__BPM_NAMES = _BPMSearch.get_names({'sec': 'SI', 'dev': 'BPM'})
+        cls.__BPM_NICKNAMES = _BPMSearch.get_nicknames(cls.__BPM_NAMES)
+        cls.__BPM_POS = _BPMSearch.get_positions(cls.__BPM_NAMES)
+
+        def _parse_nick_down_up(name, nick):
+            if '-' in nick:
```

Comment:
maybe: `sub, idx = nick.split('-') if '-' in nick else (nick, '')`

Reply:
or even better: `sub, idx, *_ = nick.split('-') + ['']`
this syntax works for all cases.

Reply:
done!